biothings/tests/biothings_pytest_plugin/biothings_pytest_plugin.py

```diff
@@ -0,0 +1,246 @@
+        raise TypeError(" └─ Invalid path or non-dictionary value encountered")
+
+# Function to fetch build version from the Hub
+def fetch_build_version_hub():
+    print("Fetching build version from the Hub...")
+    try:
+        metadata_path = os.getenv('APPLICATION_METADATA_PATH')
+        response = requests.get(metadata_path)
+        response.raise_for_status()
+        build_version_field = os.getenv('APPLICATION_METADATA_FIELD')
+        build_version_hub = get_nested_build_version_field_value(response.json(), build_version_field)
+        print(f" └─ BUILD_VERSION_HUB={build_version_hub}")
+        return build_version_hub
+    except requests.exceptions.RequestException as e:
+        print(f" └─ Failed to fetch build version from HUB. Error: {str(e)}")
+        return ""
+
+# Function to store new build version to S3
+def store_build_version_s3(build_version_hub):
+    print("Storing build version to the S3...")
+    build_version_file = f"{os.getenv('APPLICATION_NAME')}.txt"
+    if build_version_hub:
+        with open(build_version_file, 'w') as file:
+            file.write(build_version_hub)
+
+        s3 = boto3.client('s3')
+        try:
+            s3.upload_file(build_version_file, os.getenv('AWS_S3_BUCKET'), build_version_file)
+            print(" └─ Stored new build version to S3.")
+        except Exception as e:
+            print(f" └─ Failed to store build version in S3. Error: {str(e)}")
+    else:
+        print(" └─ No valid build version found, not storing to S3.")
+
+@pytest.hookimpl(tryfirst=True)
+def pytest_collection_modifyitems(config, items):
+    """Skip all tests if certain conditions are met."""
+    build_version_s3 = fetch_build_version_s3()
+    build_version_hub = fetch_build_version_hub()
+    github_event_name = os.getenv("GITHUB_EVENT_NAME", "")
+    os.environ["SEND_SLACK_NOTIFICATION?"] = "True"
+
+    if build_version_hub == build_version_s3 and github_event_name != "workflow_dispatch":
+        os.environ["SEND_SLACK_NOTIFICATION?"] = "False"
+        print("No need to run the tests.")
+        print(" └─ The S3 and Hub build versions are the same.")
+        # Skip all tests
+        for item in items:
+            item.add_marker(pytest.mark.skip(reason="Skipped due to matching build versions"))
+    else:
+        # Store new build version if tests are going to run
+        store_build_version_s3(build_version_hub)
+
+# Hook to run pytest and send Slack notification
+@pytest.hookimpl(tryfirst=True)
+def pytest_terminal_summary(terminalreporter: TerminalReporter, exitstatus: int, config):
+    """Customize pytest terminal summary and send to Slack."""
+
+    if os.getenv("SEND_SLACK_NOTIFICATION?") == "True":
+        SLACK_WEBHOOK_URL = os.getenv("SLACK_WEBHOOK_URL")
+        SLACK_CHANNEL = os.getenv("SLACK_CHANNEL")
+        SLACK_USERNAME = os.getenv("APPLICATION_NAME")
+
+        # Collect test summary information
+        total_tests = terminalreporter.stats.get('passed', []) + \
```

Comment:
recommend running black on the code for consistent formatting. This usually would be structured like the following:
```
variable = (
    <value0>
    <operation> <value1>
    <operation> <value2>
    <operation> <value3>
    ...
    <operation> <valueN>
)
```    

```
  163         # Collect test summary information
~ 164         total_tests = (
~ 165             terminalreporter.stats.get("passed", [])
~ 166             + terminalreporter.stats.get("failed", [])
~ 167             + terminalreporter.stats.get("error", [])
+ 168             + terminalreporter.stats.get("skipped", [])
+ 169         )
```